internal/app/apid/main.go

```diff
@@ -150,6 +163,10 @@ func Main() {
 						proxy.WithStreamedDetector(router.StreamedDetector),
 					)),
 			),
+
+			// TODO needed?
+			factory.WithUnaryInterceptor(injector.UnaryInterceptor()),
+			factory.WithStreamInterceptor(injector.StreamInterceptor()),
```

Comment:
I wasn't able to trigger that code path. When apid uses UNIX socket?

Reply:
The easiest way to trigger that is to do `talosctl health`.

Health check runs on the server, so it reaches out to _other_ apid nodes via its own local apid.

Path is like that: `talosctl health` -> `apid` -> `machined` -> `apid (socket)` -> other `apid`